src/negotiation/taker_negotiator.ts

```diff
@@ -47,25 +53,37 @@ export class TakerNegotiator {
   }
 
   private readonly comitClient: ComitClient;
-  private readonly makerNegotiator: MakerClient;
+  private readonly makerClient: MakerClient;
 
   constructor(comitClient: ComitClient, makerUrl: string) {
     this.comitClient = comitClient;
-    this.makerNegotiator = new MakerClient(makerUrl);
-  }
-
-  public async getOrderByTradingPair(tradingPair: string): Promise<Order> {
-    return this.makerNegotiator.getOrderByTradingPair(tradingPair);
+    this.makerClient = new MakerClient(makerUrl);
   }
 
-  public async takeOrder(order: Order): Promise<Swap | undefined> {
-    const executionParams = await this.makerNegotiator.getExecutionParams(
-      order
+  /**
+   * @description Get an order from the maker based on specified criteria. Whatever is returned from the maker is
+   * returned here, even if it does not match the criteria or is invalid. Not all criteria are passed to the maker.
+   * If it is indeed invalid or mismatching it will not be possible to execute the order, however it gives the
+   * opportunity to the lib consumer to know that this maker returns invalid orders and the details of such order.
+   * @param criteria - The criteria of the order that we are looking for.
+   */
+  public async getOrder(criteria: TakerCriteria): Promise<Order> {
+    const tradingPair = takerCriteriaToTradingPair(criteria);
+    const orderParams = await this.makerClient.getOrderByTradingPair(
+      tradingPair
     );
-    if (!executionParams) {
-      return;
-    }
 
+    return new Order(orderParams, criteria, this.execAndTakeOrder.bind(this));
+  }
+
+  /**
+   * @description **Note: This should not be used, `Order.take()` should be preferred.**
```

Comment:
🔧 if not preferred, can we hide this more properly?

Reply:
Can you please propose a way to "hide it more properly"?